### PR TITLE
Admin API proto definitions

### PR DESCRIPTION
**Pull request by codingllama**

### CODE REVIEW

**Comment by codingllama** [comment]
Issue #310

**Comment by codingllama** [comment]
Took me a while, but I figured out the problem. Travis is using protoc 3.0.0, but we need at least 3.1.0 to be able to use field_mask.proto.

I'll bump protoc on a separate PR, then rebase this one.

**Comment by daviddrysdale** [comment]
cf. #339

**Comment by codingllama** [comment]
Rebased and green: ready for review.

File: trillian_api.proto [hunk 1]
```diff
@@ -258,3 +259,69 @@ service TrillianMap {
...
+service TrillianAdmin {
+  // Lists all trees the requester has access to.
+  rpc ListTrees(ListTreesRequest) returns(ListTreesResponse) {}
+
+  // Retrieves a tree by ID.
+  rpc GetTree(GetTreeRequest) returns(Tree) {}
+
+  // Creates a new tree.
+  // System-generated fields are not required and will be ignored if present,
+  // eg: tree_id, create_time and update_time.
```
Comment (AlCutter) [inline trillian_api.proto]: e.g.

Comment (codingllama) [inline trillian_api.proto]: Done.

**Pull request merged**